Issue #2780 fix

Krishna-mishra-26 opened the pull request:
This pull request includes significant changes to the `hack/check_pre_req.sh` and `scripts/check_pre_req.sh` scripts, as well as the addition of a new `sysctl-check.yaml` file. The main focus of these changes is to streamline the process of checking system prerequisites using a Kubernetes job.

Changes to script functionality:

* `hack/check_pre_req.sh` and `scripts/check_pre_req.sh`: Replaced the existing logic for checking installed programs and sysctl values with a Kubernetes job that checks sysctl values. This includes removing the detailed checks for various programs and simplifying the scripts to create and manage the Kubernetes job.

Addition of Kubernetes job configuration:

* [`sysctl-check.yaml`](diffhunk://#diff-ae773373ef5e260c3e18525cebd95201e6244cf5a8d1ee6cc0a23e594a30d159R1-R14): Added a new Kubernetes job configuration file that defines a job to check the `fs.inotify.max_user_watches` and `fs.inotify.max_user_instances` sysctl values using a `busybox` container.

MikeSpreitzer commented:
The existing logic for checking various prerequisites depending on command line arguments is deliberate. For example, using KubeStellar does not require using `kind` (https://docs.kubestellar.io/unreleased-development/direct/pre-reqs/#software-prerequisites-for-using-kubestellar).

A user running this script does not necessarily have a Kubernetes cluster yet.

We have a design rule that requires a copyright statement at the start of every script.

MikeSpreitzer commented:
/ok-to-test

Krishna-mishra-26 commented:
This pull request focuses on simplifying the `check_pre_req.sh` script by removing complex checks and replacing them with a Kubernetes job to validate sysctl settings. The most important changes include the removal of detailed version checks for various tools and the addition of a Kubernetes job to handle sysctl checks.

### Simplification of `check_pre_req.sh` script:

* `hack/check_pre_req.sh` and `scripts/check_pre_req.sh`: Removed detailed version checks for multiple tools and replaced them with a Kubernetes job to check sysctl settings.

### Addition of Kubernetes job for sysctl checks:

* [`sysctl-check.yaml`](diffhunk://#diff-ae773373ef5e260c3e18525cebd95201e6244cf5a8d1ee6cc0a23e594a30d159R1-R14): Added a new Kubernetes job configuration to check sysctl settings (`fs.inotify.max_user_watches` and `fs.inotify.max_user_instances`).

MikeSpreitzer commented:
(A) This PR needs to be rebased onto the `main` branch.

(B) In doing so you will discover that we have deleted the copy of the script in `hack/`, this PR only needs to update the one copy in `scripts/`.

(C) We do NOT want removal of the flexibility of this script to check a collection of things specified on the command line.

(D) The user of this script generally does not already have a Kubernetes cluster available, so the script must not assume such a thing.

(E) Please do not remove the copyright statement from the script. Those are intended to be there.
